aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -255,10 +255,10 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
 
   // If dst is contiguous and there is no byte offset, we can save directly the result of
   // gather into dst. This reduces the overhead of doing an additional blit for most cases
-  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset);
+  bool returnGatherOutput = (dst_.is_contiguous() && !dst_byte_offset && src_.dtype() == dst_.dtype());
   Tensor src;
 
-  if (!src_.is_contiguous()) {
+  if (src_.is_view() || !src_.is_contiguous()) {
```

Comment:
Same as above, offset almost always indicates a view, but views can be contiguous. IMO in all those function `is_strided_contiguous` predicate should be used to determine wether src or dst tensors are backed by a contiguous non-overlapping set.
```suggestion
  if (!is_strided_contiguous(src_)) {
```

Reply:
For contiguous views with storage offset we don't need to gather the data in here - at the end of `copy_kernel_mps`, it will blit the src directly in dst at the specified storage offset